src/OpenLoco/src/GameCommands/Road/RemoveRoad.cpp

```diff
@@ -0,0 +1,262 @@
+                    setErrorText(StringIds::stringid_local_authority_wont_allow_removal_in_use);
+                    return FAILURE;
+                }
+            }
+            */
+        }
+
+        currency32_t totalRemovalCost = 0;
+
+        // 0x004776E3
+        if (auto* roadEl = getRoadElement(tile, args, args.sequenceIndex, flags); roadEl != nullptr)
+        {
+            auto* nextEl = roadEl->next();
+            auto* nextRoad = nextEl->as<World::RoadElement>();
+            if (nextRoad != nullptr)
+            {
+                auto* elStation = tile.roadStation(nextRoad->roadId(), nextRoad->rotation(), nextRoad->baseZ());
+                if (elStation != nullptr)
+                {
+                    RoadStationRemovalArgs srArgs = {};
+                    srArgs.pos = args.pos;
+                    srArgs.rotation = args.rotation;
+                    srArgs.roadId = args.roadId;
+                    srArgs.index = args.sequenceIndex;
+                    srArgs.roadObjectId = args.objectId;
+
+                    auto stationRemovalRes = GameCommands::doCommand(srArgs, flags);
+                    if (stationRemovalRes == FAILURE)
+                    {
+                        return FAILURE;
+                    }
+
+                    totalRemovalCost += stationRemovalRes;
+                }
+            }
+        }
+
+        // 0x004777EB
+        if (auto* roadEl = getRoadElement(tile, args, args.sequenceIndex, flags); roadEl != nullptr)
+        {
+            bool roadHadBridge = false; // 0x0112C2CD
+            int8_t roadBridgeId = -1;   // 0x0112C2D0
+
+            const auto& roadPieces = World::TrackData::getRoadPiece(roadEl->roadId());
+            const auto& currentPart = roadPieces[roadEl->sequenceIndex()];
+            const auto roadStart = args.pos - World::Pos3{ Math::Vector::rotate(World::Pos2{ currentPart.x, currentPart.y }, args.rotation), currentPart.z };
+
+            // NB: moved out of the loop below (was at 0x00477A10)
+            const currency32_t pieceRemovalCost = roadRemoveCost(args, roadPieces[0], roadStart, flags);
+
+            for (auto& piece : roadPieces)
+            {
+                const auto roadLoc = roadStart + World::Pos3{ Math::Vector::rotate(World::Pos2{ piece.x, piece.y }, args.rotation), piece.z };
+
+                if (!(flags & Flags::aiAllocated))
+                {
+                    World::TileManager::mapInvalidateTileFull(roadLoc);
+                }
+
+                // 0x00477934
+                auto pieceTile = World::TileManager::get(roadLoc.x, roadLoc.y);
+                auto* roadElPiece = getRoadElement(pieceTile, args, piece.index, flags);
+                if (roadElPiece == nullptr)
+                {
+                    continue;
+                }
+
+                // 0x004779B2
+                if (roadElPiece->hasBridge())
+                {
+                    roadHadBridge = true;
```

Comment:
it looks like vanilla was going to do the bridge cost only if there were no junctions on the bridge (because the bridge would still be there)

Reply:
What's the best way to check if a junction exists? Just look for a different road element (rotation) on the same tile?

Reply:
> What's the best way to check if a junction exists? Just look for a different road element (rotation) on the same tile?

well remember tram as well that could be same rotation but different road object id.